### PR TITLE
Add new source term functionality

### PR DESCRIPTION
**Pull request by timfelle**
This pull request adds new functionality to the fluid source term handler and the scalar source term handler. It includes the addition of a new procedure to append a new source term to the source_terms array in both modules. This allows for more flexibility in handling source terms in the code. Additionally, some formatting and unused variable removal changes were made.

It seem a bit silly that there are so much code duplication between the scalar and fluid source term handler. Should we maybe do a generic handler?

### CODE REVIEW

File: src/fluid/fluid_source_term.f90 [hunk 1]
```diff
@@ -189,6 +192,21 @@ subroutine fluid_source_term_free(this)
...
+  !! @param source_term The source term to be added.
+  subroutine fluid_source_term_add_source_term(this, source_term)
+    class(fluid_source_term_t), intent(inout) :: this
+    class(source_term_t), intent(in) :: source_term
+
+    integer :: n_sources
+
+    n_sources = size(this%source_terms)
+    allocate(this%source_terms(n_sources + 1))
+    this%source_terms(n_sources + 1)%source_term = source_term
```
Comment (njansson) [inline src/fluid/fluid_source_term.f90:205]: Wouldn't this remove the old sources in `this%source_terms`, unless one change to `move_alloc`

File: src/scalar/scalar_source_term.f90 [hunk 2]
```diff
@@ -180,14 +179,28 @@ subroutine scalar_source_term_free(this)
...
+  !! @param source_term The source term to be added.
+  subroutine scalar_source_term_add_source_term(this, source_term)
+    class(scalar_source_term_t), intent(inout) :: this
+    class(source_term_t), intent(in) :: source_term
+
+    integer :: n_sources
+
+    n_sources = size(this%source_terms)
+    allocate(this%source_terms(n_sources + 1))
+    this%source_terms(n_sources + 1)%source_term = source_term
```
Comment (njansson) [inline src/scalar/scalar_source_term.f90:192]: Wouldn't this remove the old sources in `this%source_terms`, unless one change to `move_alloc`

File: src/fluid/fluid_source_term.f90 (hunk 1 above)
Comment (timfelle) [reply src/fluid/fluid_source_term.f90:205]: You are probably right yes. One moment

**Review by njansson — APPROVED** [review]

**Review by timofeymukha — APPROVED** [review]

**Comment by timfelle** [comment]
@njansson we are just missing the auto-merge now.

**Pull request merged**